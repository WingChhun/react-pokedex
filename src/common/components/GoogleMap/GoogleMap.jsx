import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Map, GoogleApiWrapper, Marker } from 'google-maps-react';
import { API } from '../../constants';

function LoadingContainer(props) {
  return <div>Loading Map...</div>;
}

function GoogleMap({ locations, google, mapProps }) {
  return (
      <Map google={google} {...mapProps}>
        {locations.length &&
          locations.map((location, i) => {
            const [lat, lng] = location.split(',');

            return (
              <Marker
                key={`lat__${i}`}
                position={{
                  lat,
                  lng
                }}
              />
            );
          })}
      </Map>
  );
}

GoogleMap.propTypes = {
  locations: PropTypes.array.isRequired,
  mapProps: PropTypes.object
};

GoogleMap.defaultProps = {
  locations: []
};

export default GoogleApiWrapper({
  apiKey: API.GOOGLE,
  LoadingContainer: LoadingContainer
})(GoogleMap);
