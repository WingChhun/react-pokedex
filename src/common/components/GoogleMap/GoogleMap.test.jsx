import React from 'react';
import GoogleMap from './GoogleMap';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

//todo: test hooks
describe(`<GoogleMap/>`, () => {
  it(`renders properly`, () => {
    const wrapper = shallow(<GoogleMap />);
    expect(wrapper).toMatchSnapshot();
  });
});
