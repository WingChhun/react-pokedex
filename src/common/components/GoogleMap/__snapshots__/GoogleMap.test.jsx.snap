// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleMap/> renders properly 1`] = `ShallowWrapper {}`;
